### PR TITLE
Support for hot source reloading in the Linux glfw sky_shell

### PR DESCRIPTION
**Pull request by johnmccutchan**
- [x] Wire up synchronous script reloading in the library tag handler.
- [x] Allow glfw sky_shell to start from dart source as well as bundle.

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@jason-simmons @abarth

File: flutter/tonic/dart_library_loader.cc [hunk 1]
```diff
@@ -200,19 +200,90 @@ DartLibraryLoader::DartLibraryLoader(DartState* dart_state)
 DartLibraryLoader::~DartLibraryLoader() {
 }
 
+
+static void BlockWaitingForDependencies(
+    DartLibraryLoader* loader,
+    const std::unordered_set<DartDependency*>& dependencies) {
+  {
```
Comment (abarth) [inline flutter/tonic/dart_library_loader.cc]: This brace looks redundant.

File: sky/shell/platform/glfw/window_impl.cc [hunk 2]
```diff
@@ -99,6 +99,10 @@ void WindowImpl::RunFromBundle(const std::string& script_uri,
   engine_->RunFromBundle(script_uri, bundle_path);
 }
 
+void WindowImpl::RunFromFile(const std::string& file) {
+  engine_->RunFromFile(file, "", "");
```
Comment (abarth) [inline sky/shell/platform/glfw/window_impl.cc]: What about the rest of these arguments?

**Comment by abarth** [comment]
LGTM

File: flutter/tonic/dart_library_loader.cc (hunk 1 above)
Comment (johnmccutchan) [inline flutter/tonic/dart_library_loader.cc]: Done.

File: sky/shell/platform/glfw/window_impl.cc (hunk 2 above)
Comment (johnmccutchan) [inline sky/shell/platform/glfw/window_impl.cc]: Added a TODO for asset bundles when running from source.

**Pull request merged**